Gems/ROS2/Code/Source/Frame/ROS2FrameComponent.cpp

```diff
@@ -0,0 +1,217 @@
+
+        if (m_publishTransform)
+        {
+            AZ_TracePrintf(
+                "ROS2FrameComponent",
+                "Setting up %s transfrom between parent %s and child %s to be published %s",
+                IsDynamic() ? "dynamic" : "static",
+                GetParentFrameID().data(),
+                GetFrameID().data(),
+                IsDynamic() ? "continuously to /tf" : "once to /tf_static");
+
+            m_ros2Transform = AZStd::make_unique<ROS2Transform>(GetParentFrameID(), GetFrameID(), IsDynamic());
+            if (IsDynamic())
+            {
+                AZ::TickBus::Handler::BusConnect();
+            }
+            else
+            {
+                m_ros2Transform->Publish(GetFrameTransform());
+            }
+        }
+    }
+
+    void ROS2FrameComponent::Deactivate()
+    {
+        if (m_publishTransform)
+        {
+            if (IsDynamic())
+            {
+                AZ::TickBus::Handler::BusDisconnect();
+            }
+            m_ros2Transform.reset();
+        }
+    }
+
+    void ROS2FrameComponent::OnTick([[maybe_unused]] float deltaTime, [[maybe_unused]] AZ::ScriptTimePoint time)
+    {
+        m_ros2Transform->Publish(GetFrameTransform());
+    }
+
+    AZStd::string ROS2FrameComponent::GetGlobalFrameName() const
+    {
+        // TODO - parametrize this (typically: "odom", "world" and sometimes "map")
+        return ROS2Names::GetNamespacedName(GetNamespace(), AZStd::string("odom"));
+    }
+
+    bool ROS2FrameComponent::IsTopLevel() const
+    {
+        return GetGlobalFrameName() == GetParentFrameID();
+    }
+
+    bool ROS2FrameComponent::IsDynamic() const
+    { // TODO - determine by joint type
+        return IsTopLevel();
+    }
+
+    const ROS2FrameComponent* ROS2FrameComponent::GetParentROS2FrameComponent() const
+    {
+        return Internal::GetFirstROS2FrameAncestor(GetEntity());
+    }
+
+    const AZ::Transform& ROS2FrameComponent::GetFrameTransform() const
+    {
+        auto* transformInterface = Internal::GetEntityTransformInterface(GetEntity());
+        if (GetParentROS2FrameComponent() != nullptr)
+        {
+            return transformInterface->GetLocalTM();
+        }
+        return transformInterface->GetWorldTM();
+    }
+
+    AZStd::string ROS2FrameComponent::GetParentFrameID() const
+    {
+        if (auto parentFrame = GetParentROS2FrameComponent(); parentFrame != nullptr)
+        {
+            return parentFrame->GetFrameID();
+        }
+        return GetGlobalFrameName();
+    }
+
+    AZStd::string ROS2FrameComponent::GetFrameID() const
+    {
+        return ROS2Names::GetNamespacedName(GetNamespace(), m_frameName);
+    }
+
+    void ROS2FrameComponent::SetFrameID(const AZStd::string& frameId)
+    {
+        m_frameName = frameId;
+    }
+
+    AZStd::string ROS2FrameComponent::GetNamespace() const
+    {
+        auto parentFrame = GetParentROS2FrameComponent();
+        AZStd::string parentNamespace("");
```

Comment:
pessimiation here with the initialization of the parentNamespace variable.
The default constructor for AZStd::string is an empty string.
While this is calling the `const char*` constructor which has to perform a extra check of the `const char*` to find it's NUL terminating character.

```suggestion
        AZStd::string parentNamespace;
```